tests/communications/test_notification_preferences.py

```diff
@@ -0,0 +1,156 @@
+        self.assertFalse(pref.enable_email)
+        self.assertTrue(pref.enable_push)
+
+    def test_update_notification_type_preference(self):
+        """Test updating a specific notification type preference."""
+        pref = UserNotificationPreference.objects.get(user=self.user)
+        type_pref = pref.type_preferences.first()
+        
+        url = reverse(
+            'communications:user-notification-preference-update-type-preference',
+            kwargs={'pk': self.user.pk}
+        )
+        
+        data = {
+            'notification_type_id': type_pref.notification_type.id,
+            'frequency': 'weekly_summary'
+        }
+        response = self.client.patch(url, data, format='json')
+        
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['frequency'], 'weekly_summary')
+        
+        type_pref.refresh_from_db()
+        self.assertEqual(type_pref.frequency, 'weekly_summary')
+
+    def test_unauthorized_access(self):
+        """Test that unauthorized users can't access preferences."""
+        from rest_framework.test import APIClient
+        client = APIClient()
+        
+        list_url = reverse('communications:user-notification-preference-list')
+        response = client.get(list_url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED, 
+                        'List endpoint should require authentication')
+        
+        detail_url = reverse(
+            'communications:user-notification-preference-detail',
+            kwargs={'pk': self.user.pk}
+        )
+        response = client.get(detail_url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED,
+                       'Detail endpoint should require authentication')
+        
+        response = client.patch(detail_url, {}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED,
+                       'Update endpoint should require authentication')
+        
+        type_pref_url = reverse(
+            'communications:user-notification-preference-update-type-preference',
+            kwargs={'pk': self.user.pk}
+        )
+        response = client.patch(type_pref_url, {}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED,
+                       'Type preference update should require authentication')
+
+    def test_initial_preferences_created(self):
+        """Test that initial preferences are created for new users."""
+        new_user = User.objects.create_user(
+            email='new@example.com',
+            password='testpass123',
+            first_name='New',
+            last_name='User'
+        )
+        
+        self.assertTrue(hasattr(new_user, 'notification_preferences'))
+        self.assertEqual(
+            new_user.notification_preferences.type_preferences.count(),
+            NotificationType.objects.count()
+        )
```

Comment:
assertEqual(new_user.notification_preferences.type_preferences.count(), NotificationType.objects.count()) assumes all notification types are active and counted. If NotificationType filtering changes (e.g., is_active=False for some types), this check may not match intended business logic.